src/redux/books/books.js

```diff
@@ -0,0 +1,25 @@
+// Actions
+const ADD = 'add';
+const REMOVE = 'Remove';
+const DISPLAY = 'display';
+
+// action creator
+export const addnewBookFunction = (data) => (dispatch) => dispatch({ type: ADD, payload: data });
+export const removeBookFunction = (id) => (dispatch) => dispatch({ type: REMOVE, payload: id });
+export const displayBookFunction = () => (dispatch) => dispatch({ type: DISPLAY });
+
+// reducer function
+const bookReducer = (state = [], action) => {
+  switch (action.type) {
+    case ADD:
+      return [...state, action.payload];
+    case REMOVE:
+      return [...state, state.filter((book) => book.id !== action.payload.id)];
+    case DISPLAY:
+      return [action.payload];
```

Comment:
- [ ] [OPTIONAL] This action doesn't have any purpose since there's no `payload` passed to it from its reducer. You can remove this action as you will use something else to display the data from your state